File: 287-find-the-duplicate-number/287-find-the-duplicate-number.cpp
class Solution {
public:
    int findDuplicate(vector<int>& nums) 
    {
        map<int,int> m;
        for(int x:nums)
        {
            if(m.find(x)==m.end())
                m[x]=1;
            else
                m[x]++;
        }
        int rep;
        for(auto i:m)
        {
            if(i.second > 1)
            {
                rep = i.first;
                break;
            }
        }
        return rep;
    }
};

File: 287-find-the-duplicate-number/287-find-the-duplicate-number.java
class Solution 
{
    public int findDuplicate(int[] nums) 
    {
        HashMap<Integer, Integer> hp = new HashMap<Integer, Integer>();
        for(int i=0; i<nums.length; i++)
        {
            if(hp.containsKey(nums[i]))
                return nums[i];
            else    
                hp.putIfAbsent(nums[i], 1);
        }
        return 0;
    }
}